migrator/src/main/java/com/redhat/lightblue/migrator/ConsistencyCheckerController.java

```diff
@@ -0,0 +1,252 @@
+package com.redhat.lightblue.migrator;
+
+import java.util.Date;
+import java.util.List;
+import java.util.ArrayList;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.redhat.lightblue.client.Query;
+import com.redhat.lightblue.client.Update;
+import com.redhat.lightblue.client.Literal;
+import com.redhat.lightblue.client.request.data.DataInsertRequest;
+import com.redhat.lightblue.client.request.data.DataUpdateRequest;
+
+/**
+ * There is one consistency checker controller for each
+ * entity. Consistency checker controller is created and initialized
+ * from a MigrationConfiguration. The migration configuration gives
+ * the period with which the consistency checker runs. The consistency
+ * checker operates using a timestamp field. Every time it wakes up,
+ * it attempts to process documents that were created/modified after
+ * the last run time, and before (now-period).  When the consistency
+ * checker wakes up, it creates a migration job for every (period)
+ * length time slice.
+ *
+ * The consistency checker always leaves the last period
+ * unprocessed. This is to allow for any time discrepancies between
+ * nodes updating the database. If a node updates a record with a
```

Comment:
A minute or so of would probably suffice to cover for time discrepancies between nodes updating the database. This could be a fixed period, unrelated to the configurable checking period.